packages/math-input/src/components/keypad/mobile-keypad.tsx

```diff
@@ -34,18 +36,68 @@ type Props = {
 
 type State = {
     active: boolean;
+    containerWidth: number;
     keypadConfig?: KeypadConfiguration;
     keyHandler?: KeyHandler;
     cursor?: Cursor;
 };
 
 class MobileKeypad extends React.Component<Props, State> implements KeypadAPI {
+    _containerRef = React.createRef<HTMLDivElement>();
+    _containerResizeObserver: ResizeObserver | null = null;
+    _throttleResize = false;
     hasMounted = false;
 
     state: State = {
+        containerWidth: 0,
         active: false,
     };
 
+    componentDidMount() {
+        this._resize();
+
+        window.addEventListener("resize", this._throttleResizeHandler);
+        window.addEventListener(
+            "orientationchange",
+            this._throttleResizeHandler,
+        );
+
+        this._containerResizeObserver = new ResizeObserver(
+            this._throttleResizeHandler,
+        );
+
+        if (this._containerRef.current) {
+            this._containerResizeObserver.observe(this._containerRef.current);
+        }
+    }
+
+    componentWillUnMount() {
+        window.removeEventListener("resize", this._throttleResizeHandler);
+        window.removeEventListener(
+            "orientationchange",
+            this._throttleResizeHandler,
+        );
+        this._containerResizeObserver?.disconnect();
+    }
+
+    _resize = () => {
+        const containerWidth = this._containerRef.current?.clientWidth || 0;
+        this.setState({containerWidth});
+    };
+
+    _throttleResizeHandler = () => {
+        if (this._throttleResize) {
+            return;
+        }
+
+        this._throttleResize = true;
+
+        setTimeout(() => {
+            this._resize();
+            this._throttleResize = false;
+        }, 100);
```

Comment:
**question**: I suspect this is correct, but I got to thinking about how we want throttling to work. This approach  means we'll resize no more than once per 100ms. The other way is that we throttle so that we resize after 100ms of the last call. I think the second way would mean we visually update more slowly (wait till all calls settle for at least 100ms). Like I said, I think this is right... but I thought about it while I read the code and thought I'd mention this.

Reply:
I just stole the code from StackOverflow, but it's working in a way that I like 🤷